lib/widgets/checkout-dialog.coffee

```diff
@@ -0,0 +1,66 @@
+# coffeelint: disable=max_line_length
+
+# TODO: fix arrow down and up don't select options in focused branches select
+
+{$, View} = require 'atom-space-pen-views'
+
+module.exports =
+class CommitDialog extends View
+  @content: ->
+    @div class: 'dialog context-git', keyup: 'keyup', =>
+      @div class: 'heading', =>
+        @i class: 'icon x clickable', click: 'cancel'
+        @strong 'Checkout'
+      @div class: 'body', =>
+        @label 'Branches'
+        @select class: 'branches', outlet: 'branches'
+      @div class: 'buttons', =>
+        @button class: 'active', click: 'checkout', =>
+          @i class: 'icon desktop-download'
+          @span 'Checkout'
+        @button click: 'cancel', =>
+          @i class: 'icon x'
+          @span 'Cancel'
+
+  activate: (branches) ->
+    @branches.html(branches.map( (branch) ->
+      $option = $("<option />").attr({value: branch.name}).text(branch.path)
+      $option.attr('selected','selected') if branch.selected
+      return $option
+    ))
+    @show()
+    @branches.focus()
+    return new Promise (resolve, reject) =>
+      @resolve = resolve
+      @reject = reject
+      return
+
+  deactivate: ->
+    @modalPanel.destroy()
+    @detach()
+    return
+
+  keyup: (event, dialog) ->
+    @cancel() if event.keyCode == 27
+    @checkout() if event.keyCode == 13
+    @openBranches() if event.keyCode == 40 || event.keyCode == 38
+    return
```

Comment:
ya i'm not 100% on that either. That is one reason it is still in coffee script.